src/SFA.DAS.RoATPService.Application/Commands/PostOrganisation/PostOrganisationCommandHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.RoATPService.Application.Common.Models;
+using SFA.DAS.RoATPService.Application.Interfaces;
+using SFA.DAS.RoATPService.Application.Mediatr.Behaviors;
+using SFA.DAS.RoATPService.Domain;
+using SFA.DAS.RoATPService.Domain.Entities;
+using SFA.DAS.RoATPService.Domain.Repositories;
+using OrganisationStatus = SFA.DAS.RoATPService.Domain.Common.OrganisationStatus;
+using ProviderType = SFA.DAS.RoATPService.Domain.Common.ProviderType;
+
+namespace SFA.DAS.RoATPService.Application.Commands.PostOrganisation;
+
+public class PostOrganisationCommandHandler(IOrganisationsRepository organisationsRepository, ITextSanitiser textSanitiser) : IRequestHandler<PostOrganisationCommand, ValidatedResponse<SuccessModel>>
+{
+    public async Task<ValidatedResponse<SuccessModel>> Handle(PostOrganisationCommand command, CancellationToken cancellationToken)
+    {
+        var organisationId = Guid.NewGuid();
+
+        var auditRecord = new Audit
+        {
+            OrganisationId = organisationId,
+            UpdatedBy = command.RequestingUserId,
+            UpdatedAt = DateTime.UtcNow,
+            AuditData = new AuditData
+            {
+                FieldChanges = [],
+                OrganisationId = organisationId,
+                UpdatedAt = DateTime.UtcNow,
+                UpdatedBy = command.RequestingUserId
+            }
+        };
+
+        var isSupportingProvider = command.ProviderType.ToString() == ProviderType.Supporting.ToString();
```

Comment:
Do we need to care about casing here? Or is the command alwyas gonna come in correctly cased?

Reply:
no it's always Pascal

Reply:
But why ToString when they are both same type of enum?